# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1245142650 -7200
#      Tue Jun 16 10:57:30 2009 +0200
# Branch release-3-2-x
# Node ID 8c2a1c876c2ca1a080e2e0d911ae7f3c343383fe
# Parent  1ffb54887c425efc5ae3d616ed940fd038999a6a
fix context restoration in error state

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2009-06-16  Jaroslav Hajek  <highegg@gmail.com>
+
+	* symtab.h (symbol_table::set_scope_and_context): Avoid checking
+	error_state for setting context.
+
 2009-06-12  Kai NODA  <nodakai@gmail.com>
 
 	* ls-mat4.h: Fix include guard
 	* ov-type-conv.h: Add missing include guard.
 
 2009-06-12  Jaroslav Hajek  <highegg@gmail.com>
 
 	* ov-cell.cc (octave_cell::sort (Array<octave_idx_type>&)): Simplify.
diff --git a/src/symtab.h b/src/symtab.h
--- a/src/symtab.h
+++ b/src/symtab.h
@@ -934,21 +934,22 @@ public:
 
 	    if (p == all_instances.end ())
 	      error ("scope not found!");
 	    else
 	      {
 		instance = p->second;
 
 		xcurrent_scope = scope;
+
+                xcurrent_context = context;
 	      }
 	  }
-
-	if (! error_state)
-	  xcurrent_context = context;
+        else
+          xcurrent_context = context;
       }
   }
 
   static void set_parent_scope (scope_id scope)
   {
     xparent_scope = scope;
   }
 
